### DISCUSSION TITLE
How to test with https locally?

### DISCUSSION THREAD

**Discussion by bebraw** [skymethod/denoflare#13] [Q&A] [answered]
In order to secure my personal API, I'm using basic auth (i.e. just `Authentication` header) over https to secure it. You can find my current implementation at https://github.com/jsterlibs/jster-cf-api .

The question is, is it possible to run Denoflare with some dummy certificate at localhost? I know Deno is capable of this at least.

As a workaround I could add some kind of a development flag to disable the path when testing locally. It's not a big deal either as the current setup is nice.

**Comment by johnspurlock-skymethod** [comment, marked as answer]
Not yet, but should be straightforward to add.  Two new args to `serve` (and associated config properties) to specify the cert pem and the key pem if https is desired.

Would this work for you?

**Reply by bebraw** [reply]
Definitely.

**Reply by johnspurlock-skymethod** [reply]
Alright, just implemented it: https://github.com/skymethod/denoflare/commit/98e91be89e88a9bfa350f8c50372ca87aa6efa00

You can either specify them in the config file (via `localCertPem` or `localKeyPem` script config props) or via the command line (via `--cert-pem` or `--key-pem`).  Command line trumps.

You can install denoflare as of that commit (ie using `98e91be89e88a9bfa350f8c50372ca87aa6efa00` instead of a release tag) to try it out.  I tested it with certs generated with the [mkcert](https://github.com/FiloSottile/mkcert) tool and it worked as expected in a browser.

**Reply by bebraw** [reply]
Thanks a lot, that works perfectly.

I documented the process a bit at https://github.com/jsterlibs/jster-cf-api#configuring-stargazers in case you want to pick something for the docs. I imagine it's good to show the exact commands for people and maybe discuss the worker side a bit (what checks to make etc.) as many don't know these things well. 👍
